### ISSUE TITLE
Question: function_parameter_count rule behavior

### ISSUE THREAD

**Issue by varunpm1** [realm/SwiftLint#2171]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Possible Bug

Swift Lint rule - **function_parameter_count**

```func test(a: Int, b: Int, c: Int, d: Int, e: Int, f: Int = 0, g: Int = 0)```

The total parameters count for the above function is 7 (i.e., 5 compulsory parameters and 2 optional parameters). So I would like to know if the rule **function_parameter_count** should throw a warning for this function or not. Currently, this rule doesn't throw a warning for such functions.

```func test(a: Int, b: Int, c: Int, d: Int, e: Int, f: Int,  g: Int)```

However, making all parameters as compulsory throws warning as expected.

* Which Xcode version are you using (check `xcode-select -p`)? - XCode 9.3

As per my understanding, this should throw a warning in both the scenarios since total parameter count exceeds the limit. Any help/suggestions appreciated here.

Thanks, Varun

**Comment by varunpm1** [comment]
Understood. I think this could be added as a configuration option. Anyway, this is just my suggestion.

Thanks, Varun

**Comment by varunpm1** [comment]
Hi @marcelofabri 

I could try but I'm not sure on the steps to contributing.

**Comment by jpsim** [comment]
@varunpm1 thanks for your interest in contributing! Please read our [CONTRIBUTING.md](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md) document and let us know if you have any questions.

**Comment by varunpm1** [comment]
A small concern here -

When I tried to go through the steps as per CONTRIBUTING.md, I couldn't run the following command -
```$ make docker_test```

I'm receiving the following error -
```/bin/sh: docker: command not found```

Any help here would be appreciated.

Also, I'm assuming that this has to be a separate rule, correct?

**Comment by jpsim** [comment]
Are you sure you have docker installed? Instructions to do so here: https://docs.docker.com/install/

**Comment by jpsim** [comment]
And no, to configure the `function_parameter_count` rule to not ignore default parameters, I'd recommend you add an `ignores_default_parameters` configuration option to the rule, which would default to `true` but that you could get the behavior you want by setting it to `false`.

**Comment by varunpm1** [comment]
Got it. Thanks for the help. Trying to implement the configuration.

**Comment by varunpm1** [comment]
I have sent a PR to master branch with the required configuration. I have also added a test case for function_parameter_count rule. Let me know if anything else is needed here.

**Comment by jpsim** [comment]
Thanks for taking the initiative to do this! Myself or another contributor will hopefully have time to review your PR shortly.

**Comment by varunpm1** [comment]
Closing this issue as the PR has been merged.
